Refactor existing SGX documentation and instructions for configuring DCAP attestation (#669)
oasisprotocol/docs: docs/node/run-your-node/prerequisites/set-up-trusted-execution-environment-tee.md

- **Turbo Mode**: DISABLE
- **CPU AES**: ENABLE

> To test if your settings are correct, you may use the [attestation tool]

Reviewer: Note that this tool only works for EPID currently.

Author: Ok. Moved this tool at the end of the guide (i think it makes morse sense there) and mentioned that it only supports EPID currently.

Would it be straightforward to also add support for DCAP?

can facilitate various SGX services such as launch approval, remote attestation
quote signing, etc.

Oasis node supports the (legacy) EPID and (newer) PCS attestation methods.
Following instructions differ depending on the attestation method used.

To see if your system supports PCS attestation run the following:

```bash
 cpuid -1  | grep "SGX"
 ```

and look for the following line:
```
      SGX_LC: SGX launch config supported      = true
```

If your system doesn't support the "SGX_LC: SGX launch config supported", skip to the [EPID attestation](#legacy-aesm-epid-attestation) section.

> ## AESM PCS Attestation

Reviewer: ```suggestion
## DCAP Attestation
```

> ### Ubuntu 22.04/20.04/18.04

Reviewer: Should we still show these old Ubuntu versions given that they are not really supported (old kernel, old libc, etc.)? I would remove them.

can facilitate various SGX services such as launch approval, remote attestation
quote signing, etc.

> Oasis node supports the (legacy) EPID and (newer) PCS attestation methods.

Reviewer: ```suggestion
Oasis Core supports the (legacy) EPID and (newer) DCAP attestation methods.
```

It is IAS vs. PCS (attestation services) or EPID vs. DCAP/ECDSA. Maybe DCAP is more well-known than PCS so let's switch to DCAP everyhwere.

Author: Updated

your own PCCS. The configuration file is located at `/etc/sgx_default_qcnl.conf`.

Make sure to always restart the `aesmd.service` after updating the
configuration, via:

```bash
sudo systemctl restart aesmd.service
```

#### Intel PCS

Using the Intel PCS is the simplest and most generic way, but it may be less
reliable than using your own PCCS. Some cloud providers (see the [following section](#cloud-service-providers-pccs))
also require you to use their PCCS.

To use Intel PCS update the `pccs_url` value in `/etc/sgx_default_qcnl.conf`
to the Intel PCS API URL:

```json
  //PCCS server address
  "pccs_url": "https://api.trustedservices.intel.com/sgx/certification/v4/"
```

#### Cloud Service Provider's PCCS

Some cloud providers require you to use their PCCS.

- Azure: See the [Azure documentation] for details on configuring the quote provider. The documentation
  contains an example of an Intel QPL configuration file that can be used.

- Alibaba Cloud: See the [Alibaba Cloud documentation] for details on configuring the quote provider. The
  documentation shows the required `sgx_default_qcnl.conf` changes.

> - For Other Cloud Providers: If you are using a different cloud service provider, consult their

Reviewer: ```suggestion
- Other cloud providers: If you are using a different cloud service provider, consult their
```

sudo systemctl status aesmd.service
```

#### Configuring Intel Quote Provider

> Intel Quote Provider (`libsgx-dcap-default-qpl`) needs to be configured to use

Reviewer: ```suggestion
The Intel Quote Provider (`libsgx-dcap-default-qpl`) needs to be configured in order to use
```

order to use either the Intel PCS, the PCCS of your cloud service provider, or
your own PCCS. The configuration file is located at `/etc/sgx_default_qcnl.conf`.

Make sure to always restart the `aesmd.service` after updating the
configuration, via:

```bash
sudo systemctl restart aesmd.service
```

> 

Reviewer: Remove empty line.

#### Intel PCS

Using the Intel PCS is the simplest and most generic way, but it may be less

> reliable than using your cloud service provider (or your own PCCS).

Reviewer: Also we should somehow mention that on some cloud providers (like the below) you can only use their PCCS because the VM guest doesn't have access to the platform manifest and so can't retrieve the PCK when key caching is disabled (which is the case on Azure).

Author: Reworded a bit

To use Intel PCS update the `pccs_url` value in `/etc/sgx_default_qcnl.conf`
to the Intel PCS API URL:

```json
  //PCCS server address
  "pccs_url": "https://api.trustedservices.intel.com/sgx/certification/v4/"
```

> #### Cloud service provider

Reviewer: ```suggestion
#### Cloud Service Provider's PCCS
```

If your cloud service provider provides it's own PCCS, it's recommended to use
it. Set the `"pccs_url"` value to the respective address:

- Azure: https://global.acccache.azure.net/sgx/certification/v4/

>   - See the [Azure documentation](https://learn.microsoft.com/en-us/azure/security/fundamentals/trusted-hardware-identity-management#how-do-i-use-intel-qpl-with-trusted-hardware-identity-management) for details on configuring the quote provider.

Reviewer: Collect these long links into its own section and just reference by name (see other docs for examples).

- Alibaba Cloud: https://sgx-dcap-server-vpc.[Region-ID].aliyuncs.com/sgx/certification/v3/
  - See the [Alibaba Cloud documentation](https://www.alibabacloud.com/help/en/ecs/user-guide/build-an-sgx-encrypted-computing-environment) for details on configuring the quote provider.

- For Other Cloud Providers: If you are using a different cloud service provider, consult their
specific documentation for the appropriate PCCS configuration and guidance on configuring the quote provider, or
use one of the other PCS options.

#### Own PCCS

It is also possible to run PCCS yourself. Follow [official Intel instructions](https://www.intel.com/content/www/us/en/developer/articles/guide/intel-software-guard-extensions-data-center-attestation-primitives-quick-install-guide.html) on how to setup your own PCCS.

> 

Reviewer: Remove empty line.

curl -sSL "https://download.01.org/intel-sgx/sgx_repo/ubuntu/intel-sgx-deb.key" | sudo -E apt-key add -
```

And then install the `sgx-aesm-service`, `libsgx-aesm-ecdsa-plugin`, `libsgx-aesm-quote-ex-plugin` and `libsgx-dcap-default-qpl` packages:

```bash
sudo apt update

> sudo apt install sgx-aesm-service libsgx-aesm-ecdsa-pluginn libsgx-aesm-quote-ex-plugin libsgx-dcap-default-qpl

Reviewer: ```suggestion
sudo apt install sgx-aesm-service libsgx-aesm-ecdsa-plugin libsgx-aesm-quote-ex-plugin libsgx-dcap-default-qpl
```

sudo systemctl status aesmd.service
```

> #### Configuring Intel Quote Provider

Reviewer: ```suggestion
#### Configuring the Quote Provider
```

The example above assumes the use of the newer driver which uses two devices.
For the legacy driver you need to specify `--device /dev/isgx` instead.

:::

:::tip

Make sure to use the correct docker image based on your attestation method.
For PCS use the `ghcr.io/oasisprotocol/aesmd:master-pcs` and for EPID use the
`ghcr.io/oasisprotocol/aesmd:master-epid` image.

:::

By default, the Intel Quote Provider in the docker container is configured to use the Intel PCS endpoint.
To override the Intel Quote Provider configuration within the container mount your own custom configuration using
the `volume` flag:

```bash
docker run \
  --pull always \
  --detach \
  --restart always \
  --device /dev/sgx_enclave \
  --device /dev/sgx_provision \
  --volume /var/run/aesmd:/var/run/aesmd \
  --volume /etc/sgx_default_qcnl.conf:/etc/sgx_default_qcnl.conf \
  --name aesmd \
  ghcr.io/oasisprotocol/aesmd:master-pcs
```

The default Intel Quote Provider config is available in [Intel SGX Github repository](https://github.com/intel/SGXDataCenterAttestationPrimitives/blob/master/QuoteGeneration/qcnl/linux/sgx_default_qcnl.conf).

> ## (legacy) AESM EPID attestation

Reviewer: ```suggestion
## (Legacy) EPID Attestation
```
